src/lib/transition_frontier/transition_frontier.ml

```diff
@@ -198,6 +198,29 @@ module Make (Inputs : Inputs_intf) :
 
   let breadcrumb_of_node {Node.breadcrumb; _} = breadcrumb
 
+  module Root_history = struct
+    module Queue = Hash_queue.Make (State_hash)
+
+    type t = {history: Breadcrumb.t Queue.t; capacity: int}
+
+    let create capacity =
+      let history = Queue.create ~growth_allowed:false ~size:capacity () in
+      {history; capacity}
+
+    let lookup {history; _} = Queue.lookup history
+
+    let mem {history; _} = Queue.mem history
+
+    let enqueue {history; capacity} state_hash breadcrumb =
+      match Queue.enqueue history state_hash breadcrumb with
+      | `Key_already_present -> ()
+      | `Ok ->
+          if Queue.length history > capacity then
```

Comment:
You seem to set the size of the queue equal to capacity, but then you expect the queue to grow past the capacity here. I think the call to enqueue may actually fail here since growth is not allowed. You should probably check the length before you try and add the item to the queue, or make the queue size capacity + 1.

Reply:
Doing, `Queue.create ~growth_allowed:false ~size:capacity ()`, does not actually cap the size of the queue. I think it only caps the buckets that are internally used for the Hashtable. 

You can try it on utop:

```
utop # let queue = Int.Hash_queue.create ~growth_allowed:false ~size:1 ();;
val queue : '_weak3 Int.Hash_queue.t = <abstr>
utop # Int.Hash_queue.enqueue queue 1 'a';;
- : [ `Key_already_present | `Ok ] = `Ok
utop # Int.Hash_queue.enqueue queue 2 'b';;
- : [ `Key_already_present | `Ok ] = `Ok
utop # Int.Hash_queue.to_list queue;;
- : char list = [a; b]
```

I will just remove the extra arguments to not make the confusion.

Reply:
Very odd. Ok, well if it works, it works.